python/ql/src/semmle/python/objects/ObjectAPI.qll

```diff
@@ -581,13 +581,6 @@ abstract class FunctionValue extends CallableValue {
             exists(Expr expr, AstNode origin | expr.pointsTo(this, origin) | not origin instanceof Lambda)
         )
     }
-
-    /** Gets a class that this function may return */
-    ClassValue getAnInferredReturnType() {
-        result = this.(BuiltinFunctionValue).getAReturnType()
-        or
-        result = this.(BuiltinMethodValue).getAReturnType()
-    }
```

Comment:
Ah, I'm sorry. I think I misread the code. This one is fine to stay, as long as it refers to `(BuiltInFunctionObjectInternal)` instead of `(BuiltinFunctionValue)` etc.

It was the duplication of code from `getAReturnType` that bothered me.

So, to be clear, what I propose is 
1. Removing again `CallableObjectInternal::getAnInferredReturnType` (which was a bit odd anyway, as it returned a `ClassValue`, which lives in a different part of the API).
2. Adding back the above, with `ObjectInternal` instead of `Value`
3. Using the above method instead of `(CallableObjectInternal).getAnInferredReturnType()` in `ReturnValueIgnored.ql`.

Reply:
I'm sorry, I must somehow have been unclear in what I wrote above. I can't see that _any_ of the requested changes I mentioned have been done. 

(In fact, the only new commit I see just autoformatted everything.)

Reply:
Ok, should be done now?